include/envoy/network/connection.h

```diff
@@ -116,6 +116,13 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
   virtual const std::string& remoteAddress() PURE;
 
   /**
+   * @return The address the remote client is trying to connect to.
```

Comment:
This comment is still confusing from an interface perspective and I think should read (this would also eventually allow us to use this for logging client connection outbound addresses for example):

"@return the local address of the connection. For client connections, this is the origin address. For server connections, this is the local destination address. For server connections it can be different from the proxy address if the downstream connection has been redirected or the proxy is operating in transparent mode.